core/common/src/main/java/alluxio/metrics/MetricsSystem.java

```diff
@@ -39,225 +44,251 @@
  * The syntax of the metrics configuration file is:
  * [instance].[sink|source].[name].[options]=[value]
  */
-@NotThreadSafe
-public class MetricsSystem {
+@ThreadSafe
+public final class MetricsSystem {
   private static final Logger LOG = LoggerFactory.getLogger(Constants.LOGGER_TYPE);
 
   // Supported special instance names.
   public static final String MASTER_INSTANCE = "master";
   public static final String WORKER_INSTANCE = "worker";
+  public static final String CLIENT_INSTANCE = "client";
+
+  public static final MetricRegistry METRIC_REGISTRY = initMetricRegistry();
```

Comment:
It is a final static threadsafe variable. And is used to create counters (for example, FileSystemMaster).